

# Ecological Genomics Notebook  

## Author: Erika Bueno  
### Affiliation: University of Vermont
### E-mail contact: embueno@uvm.edu


### Start Date: 2020-01-13
### End Date: 2020-05-08
### Project Descriptions:   





# Table of Contents:   
* [Entry 1: 2020-01-13, Monday](#id-section1)
* [Entry 2: 2020-01-14, Tuesday](#id-section2)
* [Entry 3: 2020-01-15, Wednesday](#id-section3)
* [Entry 4: 2020-01-16, Thursday](#id-section4)
* [Entry 5: 2020-01-17, Friday](#id-section5)
* [Entry 6: 2020-01-20, Monday](#id-section6)
* [Entry 7: 2020-01-21, Tuesday](#id-section7)
* [Entry 8: 2020-01-22, Wednesday](#id-section8)
* [Entry 9: 2020-01-23, Thursday](#id-section9)
* [Entry 10: 2020-01-24, Friday](#id-section10)
* [Entry 11: 2020-01-27, Monday](#id-section11)
* [Entry 12: 2020-01-28, Tuesday](#id-section12)
* [Entry 13: 2020-01-29, Wednesday](#id-section13)
* [Entry 14: 2020-01-30, Thursday](#id-section14)
* [Entry 15: 2020-01-31, Friday](#id-section15)
* [Entry 16: 2020-02-03, Monday](#id-section16)
* [Entry 17: 2020-02-04, Tuesday](#id-section17)
* [Entry 18: 2020-02-05, Wednesday](#id-section18)
* [Entry 19: 2020-02-06, Thursday](#id-section19)
* [Entry 20: 2020-02-07, Friday](#id-section20)
* [Entry 21: 2020-02-10, Monday](#id-section21)
* [Entry 22: 2020-02-11, Tuesday](#id-section22)
* [Entry 23: 2020-02-12, Wednesday](#id-section23)
* [Entry 24: 2020-02-13, Thursday](#id-section24)
* [Entry 25: 2020-02-14, Friday](#id-section25)
* [Entry 26: 2020-02-17, Monday](#id-section26)
* [Entry 27: 2020-02-18, Tuesday](#id-section27)
* [Entry 28: 2020-02-19, Wednesday](#id-section28)
* [Entry 29: 2020-02-20, Thursday](#id-section29)
* [Entry 30: 2020-02-21, Friday](#id-section30)
* [Entry 31: 2020-02-24, Monday](#id-section31)
* [Entry 32: 2020-02-25, Tuesday](#id-section32)
* [Entry 33: 2020-02-26, Wednesday](#id-section33)
* [Entry 34: 2020-02-27, Thursday](#id-section34)
* [Entry 35: 2020-02-28, Friday](#id-section35)
* [Entry 36: 2020-03-02, Monday](#id-section36)
* [Entry 37: 2020-03-03, Tuesday](#id-section37)
* [Entry 38: 2020-03-04, Wednesday](#id-section38)
* [Entry 39: 2020-03-05, Thursday](#id-section39)
* [Entry 40: 2020-03-06, Friday](#id-section40)
* [Entry 41: 2020-03-09, Monday](#id-section41)
* [Entry 42: 2020-03-10, Tuesday](#id-section42)
* [Entry 43: 2020-03-11, Wednesday](#id-section43)
* [Entry 44: 2020-03-12, Thursday](#id-section44)
* [Entry 45: 2020-03-13, Friday](#id-section45)
* [Entry 46: 2020-03-16, Monday](#id-section46)
* [Entry 47: 2020-03-17, Tuesday](#id-section47)
* [Entry 48: 2020-03-18, Wednesday](#id-section48)
* [Entry 49: 2020-03-19, Thursday](#id-section49)
* [Entry 50: 2020-03-20, Friday](#id-section50)
* [Entry 51: 2020-03-23, Monday](#id-section51)
* [Entry 52: 2020-03-24, Tuesday](#id-section52)
* [Entry 53: 2020-03-25, Wednesday](#id-section53)
* [Entry 54: 2020-03-26, Thursday](#id-section54)
* [Entry 55: 2020-03-27, Friday](#id-section55)
* [Entry 56: 2020-03-30, Monday](#id-section56)
* [Entry 57: 2020-03-31, Tuesday](#id-section57)
* [Entry 58: 2020-04-01, Wednesday](#id-section58)
* [Entry 59: 2020-04-02, Thursday](#id-section59)
* [Entry 60: 2020-04-03, Friday](#id-section60)
* [Entry 61: 2020-04-06, Monday](#id-section61)
* [Entry 62: 2020-04-07, Tuesday](#id-section62)
* [Entry 63: 2020-04-08, Wednesday](#id-section63)
* [Entry 64: 2020-04-09, Thursday](#id-section64)
* [Entry 65: 2020-04-10, Friday](#id-section65)
* [Entry 66: 2020-04-13, Monday](#id-section66)
* [Entry 67: 2020-04-14, Tuesday](#id-section67)
* [Entry 68: 2020-04-15, Wednesday](#id-section68)
* [Entry 69: 2020-04-16, Thursday](#id-section69)
* [Entry 70: 2020-04-17, Friday](#id-section70)
* [Entry 71: 2020-04-20, Monday](#id-section71)
* [Entry 72: 2020-04-21, Tuesday](#id-section72)
* [Entry 73: 2020-04-22, Wednesday](#id-section73)
* [Entry 74: 2020-04-23, Thursday](#id-section74)
* [Entry 75: 2020-04-24, Friday](#id-section75)
* [Entry 76: 2020-04-27, Monday](#id-section76)
* [Entry 77: 2020-04-28, Tuesday](#id-section77)
* [Entry 78: 2020-04-29, Wednesday](#id-section78)
* [Entry 79: 2020-04-30, Thursday](#id-section79)
* [Entry 80: 2020-05-01, Friday](#id-section80)
* [Entry 81: 2020-05-04, Monday](#id-section81)
* [Entry 82: 2020-05-05, Tuesday](#id-section82)
* [Entry 83: 2020-05-06, Wednesday](#id-section83)
* [Entry 84: 2020-05-07, Thursday](#id-section84)
* [Entry 85: 2020-05-08, Friday](#id-section85)

------
<div id='id-section1'/>   
### Entry 1: 2020-01-13, Monday.   

Today we discussed the definition of Ecological genomics and how it relates to the history of population genetics. Specifically, I learned about the founders of population genetics (Seawall Wright, R.A. Fisher, J.B.S. Haldane) and their contributions to the field. For example, Wright's island model is used to predict how subpopulations "represented by islands" may behave assuming population size is the same and always stable (don't grow or decline).

![Image result for wright's island model](https://www.ucl.ac.uk/~ucbhdjm/courses/b242/MigEvol/OneIsland.gif)

Notes: Key points from lecture

Fixation coefficient (Fst)- a measure differentiation among populations. 

* Fst values range from 0-1
  * High Fst value close to 1 infer that the population is highly differentiated (high variability) 
  * Values closer to 0 infer low differentiation
  * High values are associated with less migration (gene flow) and non-random mating

Selection sweeps- occurs when an allele becomes more frequent in the population- we can infer that this allele confers a beneficial trait.

Molecular evolution- "history encoded in DNA" can be examined at different levels

1. Populations
2. Organisms
3. Species

Kimura's Neutral Theory of Molecular Evolution- estimates the probability of a new mutation in a population (N)
$$
1/2N= Î¼
$$
Synonymous vs non-synonymous mutations-

* Synonymous mutations- does not alter the amino acid in the protein

* Non-synonymous mutations- alters the amino acid in the protein
  $$
  dN/dS
  $$
  N mutations/non syn. sites/ N of syn./ N of syn sites

  w= dN/dS 

  w>>1 or w>2 signal for positive selection

  w = 1 neutral

  w<< 1 negative "purifying selection"

Tajima's D-  ðœ‹ /s

ratio of nucleotide diversity ( ðœ‹ )

no. of segregating sites (s)

High  ðœ‹  with rare variants in population



------
<div id='id-section3'/>   
### Entry 2: 2020-01-15, Wednesday.   

Lecture notes: How do the questions tie in with the technology?

Things to consider: type organism/sample (tissue, environment, DNA, RNA)

Does the organism have a reference genome? How large is the genome? 

**Library preps**

1. **Whole-genome sequencing**

2. **RNAseq**

3. **Exome Capture**-  not a measurement of expression; focused on examining variation

   Exon- protein coding regions in DNA

   Exome - portion of the genome that codes for exons

   Biomedical uses- disease causing variants/rare diseases

   Ecological and evolutionary- genetic mapping of phenotypic traits, IDing causing variants, detecting selection in a genome

   Cost effective and allows one to beyond model taxa

   Information on life history strategies and molecular markers for evolution

   **Library prep:**

   1. Sample collection
   2. Fragmentation- "fragmented soup" blunt ends created, fragments same length, adenine tails added
   3. Enrichment of libraries- pull DNA and add probes 
      1. PCR- good for 100s loci, low cost, per probe cost is high
      2. De novo- good 100-1000 loci, high cost, per probe cost intermediate
      3. Divergent annotated genome- good for 100-1000s loci, cost per probe low,
   4. Purification
   5. Quality control

4. **RAD/GBS:** 

   RADseq: restriction site associated DNA seq.

   â€‹	Uses APK1- enzyme (restriction enzyme "cuts" palindromic sites) allows for barcode attachment

   Library prep:

   1. Genomic DNA
   2. Digest with 1 restriction enzyme
   3. Ligate adapters to fragments
   4. Multiplex (pool) samples
   5. PCR- allele drop-off issues-> PCR amplification does not work or cut site because of an existing variant thus reducing the amount of DNA being seq.

5. **Amplicon seq**- sequencing singular fragment of DNA- often highly conserved

   1. 16s RNA for bacteria
   2. ITS for fungi
   3. CO1 for animals

   **Library prep**

   1. DNA isolation
   2. PCR - amplify product of interest
   3. Add adapters- for barcoding
   4. Sequence

6. **Bisulfite seq: Identifies  methylated C's in genome.** 

   CG islands: GCGCGCGC - usually in non-coding DNA or promoter regions of a gene

   Library prep:

   1. Isolate DNA
   2. Treatment with bisulfite- converts methylated cytosine with a uracil
   3. PCR- amplify 
   4. Sequence treated samples
   5. Compare to untreated samples

CHIPseq: chromatin immunoprecipitation

Library prep:

1. Add fixative to fix all proteins attached to DNA
2. Immunoprecipitation: introduce an antibody that is specific to protein of interest
3. Sequence- data tells you where histone modifications are located

ATAC seq: Chromatin accesibility

"Tagementation": tagging probes + fragmenting DNA

1. Introduce a manufactured TE to fragment DNA
2. Load fragments into sequencer 
3. (HiSeq)
4. Global 3-D shape of the genome- crosslinks all touching DNA, only fragment those crosslinked DNA

**Sequencing platforms**

Illumina- for short read sequencing

sequencing: decoding nucleotide molecules

NGS- faster and cheaper

* Four step process:
  	1. Library prep: DNA fragmentation, attaching adapters
   	2. Cluster generation: lawn- contains complementary adapters + DNA polymerase + bases (ATCG) = creates clusters
   	3. Sequencing by synthesis: oligo's + fluorescent dNTPs = fluorescent signal is measured by the machine
   	4. Data analysis: filtering and mapping to reference genome

PacBio/Nanopore- long read sequencing

Long read seq: 

```mermaid
graph TD
A(<b> Long read seq)
B(<b>SMART)
C(<b>Synthetic)
A --> B
A --> C
B --> D(PacBio Oxford Nanopore Technology)
C --> E(Illumina)
C --> F(OX Genomics)

```

PacBio- small well set up where single DNA fragment foes in and polymerized

* dNTPs + fluorophores -> lasers and camera

Synthetic- wells (Illumina) vs emulsion based (OX genomics)

------
<div id='id-section8'/>   
### Entry 3: 2020-01-22, Wednesday.   

**Population genomics**

â€‹	Why is it used? allows one to infer selection and drift across the genome

-Population genetics does not incorporate the genomic landscape and structure

-The effects of environmental changes can be tracked via changes in abundance and distribution (demographics) as well as changes in selection pressures- "potential to adapt"

Challenges: teasing apart demographic history from effects of selection

Why do population genomics?

-genomic environment is inherently variable

-allows one to statistically capture the variation with dense sequencing

-population genetic approaches miss out total variation because of limited sampling

**Linkage disequilibrium**- caused by physical distance due to crossover events

1) **genomic context:** selection acts on the genome locally while drift acts on the genome globally

* mean and distribution used to infer drift and small population size

* outliers allows one to infer selection

* small population size and drift act on the entire genome

**2) Demographic history**

Ne- effective population size

* behaves as a Wright-Fisher population
* idealized population; every individual has same probability of mating
* migration (gene flow) among populations (groups of individuals exchanging genes)
* divergence - times between populations

**Coalescent theory**- probability of allele copies share a common ancestor

2N allele copies in a population of Ne individuals

1/2Ne= probability of a common ancestor

* stochastic process- sample multiple alleles copies not just one to tease out variability
* MRCA- Most recent common ancestor of all allele copies
* Population bottlenecks change the rate coalescent events 
* Rate speeds up during population bottlenecks and slows down with expansion
* inferring drift based on models of coalescence
* adding mutations to find derived alleles in population

Bottlenecks- loose rare alleles and maintain common ones

Site frequency spectrum (SFS)- histogram of allele frequencies for many SNPs in a population

3) **Selection**- positive selection increases frequency of beneficial allele/variant

* nucleotide diversity- average pairwise differences between individuals
* low nucleotide diversity under positive selection
* linkage measures- increase near selected sites due to genetic hitchhiking
* cause exact same expansion in tree "bushy trees" as bottleneck but only locally- selective sweep

```mermaid
graph TD

A(<b>Common ancestor)
B(<b> sample A)
C(<b> sample B)
A --> B
A --> C

```

------
<div id='id-section11'/>   
### Entry 4: 2020-01-27, Monday.   



------
<div id='id-section13'/>   
### Entry 5: 2020-01-29, Wednesday.   

Lecture notes: What's the difference between Rmd vs md?

* markdown a type of language that renders it into something readable

* inserting web links: [TEXT](URL)

* numbered lists: 1. TEXT

* embedding code: ``` LINE BREAK ```

  ``` cd ~/mydata```

* This is where the class data are: ` /data/project.data`
* two windows: notebook open and terminal
* pbio381.uvm.edu server name

Learning objectives

* background on species
* understand the pipeline
* bash scripts
* sequencing quality assessment
* mapping to ref genome

Red spruce- cool moist climates, northerly distributed, higher in elevation as you move south.

* genetic basis of climate adaptation
* highly fragmented populations
* alleles that might be potentially conserved
* inform nature conservancy to improve restoration management practices for red spruce

Genomic DNA from needles (110 mother trees; 23 populations)

* ref genome based on multiple tissues and developmental stages in related species white spruce
* 80,000 probes were designed based on these transcriptomes (120 bp)
* coverage of at least one probe for 38,570 unigenes (90 bp, 85% coverage)
* fragmentation, hybridization with probes, sequenced on a Hiseq X to generate paired end 150 bp reads.
  * single end only one end is paired
  * paired ends two ends are sequenced- more powerful than single end

Pipeline

1. fastq files- compressed files containing reads 
   1. often zipped in a fastq.gz
2. visualize with fastQC- assess quality
3.  trimmed .fastq - trimmomatic- visualize again with fastQC
4. map or align those reads to ref genome with BWA (.sam)
5. post processing- samtools (.bam)
   1. look for PCR duplicates- avoiding pseudoreplication by getting rid of duplicate reads
   2. generate stats around depth and quality for each sample

`cd /data/project_data/RS_ExomeSeq//fastq/edge_fastq/`

naming- population code_mother tree_read direction_file ext

compressed .gz format- cant use head code`

`zcat FILE | head -n 4` #first four lines

metadata about run

`@GWNJ-0842:368:GW1809211440:2:1101:18355:2170 2:N:0:NCGTATCA+NTTCGCCT
NACAGAATTAATTTCATGGGAGAGCAAATGTACGTAACTAGGAAAATTTAAAGGGCTGTATGCACCCATATCCTCCTCATCCTCATAAGCATCATCAACAGATATGATAGGAAGAATTATGCCAGTTACAAGCTAGAACCCTCCCTTTTT
+
#AA<7A-A-AJA--<F--7F---7--<-<----<F----<-7--<<FFJ--<F-<--7--<F7-FJ<AF-A777<<F--<FF<-F--<-7-F-AFF<-F-<A-7-77-7A<F-7<-7<-77F-F<<F<-FFFJ-------)-)))77-AF`

1. information on which machine
2. barcode sequence
3. sequence (including any N's)
4. separator is a plus sign
5. ASCI representation of the quality of the read

Phred quality scores- 0-40: what's the probability that the machine read the correct nucleotide

p = 10 ^-38/10

= 10^-3.8

p= 0.0038 # probability that the correct nucleotide was called

Running fastQC -for CRA individuals

usually the text files get redirected to local machine- for this example were redirectiong to shared space

`repo`

my scripts - vim- press i (active edit mode) - #!/bin/bash #language info 

`cd ~/EcolGenomics/myresults/ #where we are saving the fastqc results
mkdir fastqc` #making the folder where it will save

creating a for loop

`for file in /data/project_data/RS_ExomeSeq/fastq/edge_fastq/CRA*fastq.gz`

do

`fastqc ${file} -o fastqc`

done

shift + colon 

:wq fastqc.sh #bash script saved in  myscripts and exit the vim

fix permission to run the script

`chmod u+x` #an X should appear now

`bash fastqc.sh`

it should run git

git pull after

results- basic statistics

```
git pull
git add --all 
git commit -m "note"
git push
```

Trimming our reads with Trimmomatic

`cd /data/scripts`

`cp trim_loop.sh  /Ecolgenomics/myscripts/`

`vim trimm_loop.sh`

edit script for individual population

`cd /data/project_data/RS_ExomeSeq/fastq/edge_fastq

for R1 in CRA*R1_fastq.gz

do

        R2=${R1/_R1_fastq.gz/_R2_fastq.gz}
        f=${R1/_R1_fastq.gz/}
        name=`basename ${f}`
    
        java -classpath /data/popgen/Trimmomatic-0.33/trimmomatic-0.33.jar org.usadellab.trimmomatic.TrimmomaticPE \
        -threads 1 \
        -phred33 \
         "$R1" \
         "$R2" \
         /data/project_data/RS_ExomeSeq/fastq/edge_fastq/pairedcleanreads/${name}_R1.cl.pd.fq \
         /data/project_data/RS_ExomeSeq/fastq/edge_fastq/unpairedcleanreads/${name}_R1.cl.un.fq \
         /data/project_data/RS_ExomeSeq/fastq/edge_fastq/pairedcleanreads/${name}_R2.cl.pd.fq \
         /data/project_data/RS_ExomeSeq/fastq/edge_fastq/unpairedcleanreads/${name}_R2.cl.un.fq \
        ILLUMINACLIP:/data/popgen/Trimmomatic-0.33/adapters/TruSeq3-PE.fa:2:30:10 \
        LEADING:20 \
        TRAILING:20 \
        SLIDINGWINDOW:6:20 \
        MINLEN:35

escape + :wq  #escaping vim mode

`bash trim_loop.sh ` #runs the program

**HW: run fastqc on our trimmed sequences- cleaned paired reads for R1 and R2 ONLY**

location of cleaned reads:

`/data/project_data/RS_ExomeSeq/fastq/edge_fastq/pairedcleanreads/ `

------
<div id='id-section16'/>   
### Entry 16: 2020-02-03, Monday.  

Paper:

Notes:

**Fst**- a measure of genetic structure developed by Sewall Wright (1969, 1978). Related to statistical analysis of variance (*ANOVA)* 

[Source]: https://www.uwyo.edu/dbmcd/popecol/maylects/popgengloss.html

*F*ST is the proportion of the total genetic variance contained in a subpopulation (the S subscript) relative to the total genetic variance (the *T* subscript). 

* Values can range from 0 to 1. High *F*ST implies a considerable degree of differentiation among populations.

**Selective Sweeps -** When a new beneficial mutation is positively selected for and a reduction or elimination of variation is observed as a result. 

* Hard sweep- typically result from rapid emergence of mutations. 
* Soft sweeps are a result of selection on a allele that is carried by multiple individuals in a population.

**Admixture**: Gene flow between two populations that are genetically differentiated, involves gene flow in both directions. Admixture can be plotted with a population structure plot, which allows one to identify the number of genetically differentiated groups in a population and the amount of admixture between the groups.

**Tajimaâ€™s D** (aka as a neutrality test) is commonly used in population genetics to determine whether a DNA sequence evolved from random processes (mutation and genetic drift) or through non-random processes (i.e. directional selection, genetic hitchhiking, introgression, etc.). It is calculated by subtracting the difference between the observed variation and the expected variation in a population. 

* A negative Tajimaâ€™s D means that genetic variation is being reduced through population expansion (i.e. recovering from small pop size) or a selective sweep (decreases variation). 
* A positive Tajimaâ€™s D suggests that genetic variation is being maintained by balancing selection or a recent population contraction.

Info session notes: Genomic diversity 

Effective population size: size of an ideal population that conforms to Wright-Fisher assumptions.

* Factors deviating Ne from N

  * non-random mating, unequal sex ratio, selection, reproductive skew, constant population size

* Seawall Wright Ne= 4Nmales x Nfemales/ Nmales + Nfemales (founded upon this equation on sex ratios)

* can be estimated by pie = Ne*u (net probability of Ne times mutation rate)

* mutation rate-units change per time per generation

* Nei's genetic distance- measurement of diversity

* Fst- population structure- 1/ pie +1 or 1 - avg H of subpop/ H for whole pop (H - heterozygosity)

* Site frequency spectrum- number of individuals (frequency) plotted against number of differentiated SNPS (polymorphisms) 

  ![](https://s3-eu-west-1.amazonaws.com/ppreviews-plos-725668748/1170708/preview.jpg)

Frequency of rare alleles in a population. With population expansion, Ne increases, and the frequency of rare alleles will increase in a few number of individuals.

Fst = 1- pieS/pieT



------
<div id='id-section18'/>   
### Entry 18: 2020-02-05, Wednesday.  

Notes: 

Start mapping each set of cleaned reads to ref genome

* viz sequence alignments files
* process sam files
  * converting to bam  format and sorting by coords
  * removing PCR dups
  * indexing for fast future lookup

Calculate mapping statistics to assess quality of the results

Learn how to put separate bash scripts into a "wrapper" that runs them all

Ref genome: from norway spruce- close relative to red spruce

* wget- to download file remotely from internet

* created a reduced reference genome containing all contigs since genome size is too large 

* the order of the contigs are not ordered to chromosome

* N50:  how far do you go to count up to 50% of the all contigs- represents the sum of all contigs that give half of the assembled genome
  * assessment measurement of the assembly "quality"

Wrapper- all scripts together

mypipeline.sh-

mapping.sh- alignment to ref genome

process_bam.sh 

* bwa program: algorithm for mapping reads to refs

  `bwa mem -t 1 -M ${ref} ${forward} > ${output}/BWA/${name}.sam`

  

  detaching for run 

  `screen`

  `bash mypipeline.sh`

  CtrlA + ctrlD #to detach

  screen -r #to go back to screen

  

  

### Entry 21: 2020-02-10, Monday.   

GWAS- genome association studies

* First performed in 2002
* rare variants mapped to SNPs in the genome
  * e.g. diabetes, parkinsons, and heart abnormalities

Manhattan plots- SNP's mapped across the genome (chromosomes)

	 * statistically sig SNPs are shown above the log10(p-value )line
	 * spatial auto-correlations between SNP's 

Methodology- finding populations that differ in a singular traits and sequencing genome to map those traits to SNP's

1. Allows one to find causal impacts of genetic variance

2. close to causal SNPs - linkage disequilibrium- genes associated to variant due to spatial linkage on the chromosome

3. population bias - based only population sampled

   Challenges:

   SNP's are not genes

   low heritability of traits- distinctions can become unclear

   ancestral association- bigger issue when population size (population demographics) is small - less recombination- more homogeneous genetic makeup

   population structure- fitting the data; removing the variability to see if the remaining vars explain the patterns seen

   lambda=Median empirically observed/median empirically expected

   Ecological source of error- 

   â€‹	methodology is different

   â€‹	correlated traits- principal component analysis

   â€‹	trait structure extrapolation- causal SNP and phenotype;   when effective size and allele frequencies are high

   Significance threshold- competition between accuracy and broadness

   

------
<div id='id-section23'/>   
### Entry 23: 2020-02-12, Wednesday

`head CRA_01.sam` - first 10 lines in the file (name of headers followed by the length of the mapped read)

`tail CRA_01.sam` - sam format info on wiki

Quality score check on- https://broadinstitute.github.io/picard/explain-flags.html, enter the quality score i.e. 147 and check for failed mapping 

`MA_10131732` contig in the genome that the read mapped to

mapping quality is numerical phred score- higher scores are better

writing a loop to generate all statistical info on mapped reads

filename sorted.rmdup.bam  (.bai preferred input file)

`samtools tview /data/project_data/RS_ExomeSeq/mapping/BWA/CRA_01.sorted.rmdup.bam /data/project_data/RS_ExomeSeq/ReferenceGenomes/Pabies1.0-genome_reduced.fa`

ANGSD- provide SNP freq. to create a SFS and then calc diversity metrics pi, theta, and Tajimas' d

output is always a list of bam files

to check if files are there after running angsd

ll in the angsd folder



------
<div id='id-section26'/>  
<div id='id-section28'/>   
### Entry 28: 2020-02-19, Wednesday.   



folding the SFS- taking the midpoint of the bins in the SFS plot and "fold" the SFS over each other 

folded spectra: minor allele frequency (rare alleles)

Estimating rough SFS

Refine SFS using the first run as a prior

doTheta (parameter for population diversity) 

theta= 4NEu

human readable stats- theta_stats







------
<div id='id-section31'/>   
### Entry 31: 2020-02-24, Monday.   

**Ecological genomics**

Population genetics (p+q, coalescent theory) + molecular evolution  (pi, theta, Fst, SFS) + ecology (biotic and abiotic) + NGS

where does transcriptomics fit in the venn diagram? 

* measuring gene expression in a **controlled environmental** setting to reveal g**enetically controlled** differences of **phenotypes at the molecular level**
*  what is a phenotype? characteristics of organisms that can be observed
  * i.e. coat color (morphological), disease resistance/susceptibility (cellular and molecular level)
* Central dogma of molecular biology
  *  	transcription		translation                     N.S. acts on the phenotype 
  * DNA ---------> RNA ----------------> Protein = phenotype of interest
  * reverse ecology- signatures from DNA to indicate the phenotypes
  * transcriptomics: measures functional expression in route to a specific phenotype
* mutation - coding regions- protein functions- phenotype
  * splice site variation - intron exon 
  * promoter and enhancers affecting expression level in the cells and timing of expression or the cell/tissue type in which the gene is expressed, conditions of expression
  * microRNA- modifications at the RNA level
* Proteins- post translational modifications i.e. phosphorylation 
* Epigenetic modifications
* Benefits: active expression good link to phenotype, global/unbiased, illumina (cheap), de novo transcriptome

P = G + E + G*E (-control E- hold it constant)

Experimentally induce the phenotype of interest

-----> goal: molecular underpinnings of a phenotype

**Question**: what do you want to know/what's your hypothesis? this will effect the factors 

**Factors**

* common gardens : controlling the environmental condition 
* treatments or conditions: related to phenotype of interest
* what populations? 
* tissue, cell, life history stage
* transgenerational environment
* sex of the organism and reproductive stage

replication is important

3 pops x 2 types x 5 reps = 30 samples

Workflow

1. careful experimental design- 
   * question, hypothesis, broad 'omics, a priori hypothesis
   * Factors
2. Experiment
   *  sample tissues individuals
   * save RNA
     * extract, prepare, sequence- whole mRNA sequencing or 3' tag sequencing
3. Process and analyze data
   * check quality --> clean --> check quality again
   * map to reference transcriptome and extract count data (how many reads aligned to a particular gene?)
   * normalization step
   * count data matrix 
   * use factors to run statistical analysis i.e. ANOVA 
   * counts can be used to make reaction norms across different conditions 
     * expression ~ treatment + family + treatment*family
4. Integrate: GO, network analysis, identify gene correlations in expression, identify the function of correlated genes, generate diversity stats, integrate microbiome data, integrate with epigenetic and proteomic data, and environmental data





------
<div id='id-section33'/>   
### Entry 33: 2020-02-26, Wednesday.   

RNA seq dataset

purpose of this experiment was to sample red spruce genetic variation from sites that were polarized into cool & wet vs. hot and dry based on historic climate

quant.sf files generated by Salmon -> inputted in DEseq2 for analysis and viz

goal: to assess the gene expression responses of individuals from these habitats in response to experimental treatments of heat and heat+drought

10 maternal families -  5 from hot and dry and 5 from cool and wet

one year old seedlings-tissues harvested day 0, 5, and 10

RNA extracted from whole seedlings (root, stem, leaves)

aimed for 5 biol reps per treatment x sourceclim x day combo but day 5 had few RNA extractions that worked

total N = 76 samples

3' tag sequencing- uses little amount of RNA (Cornell)

Questions asked

seedlings grown in a common garden

Factors: Treatment (C, H, D)

SourceClim ( CoolWet, HeatDry)

Time (0, 5, 10 days)

1) Do different individuals differ in gene expression across treatment

2) " " at different time points

expression ~ Time + SourceClim + Treatment + Time*SourceClim + family

expression ~ Time + SourceClim + Treatment + SourceClim*Treatment +family 

expression ~ Time + SourceClim + Treatment + SourceClim*Treatment *Time +family 

3 factor ANOVA (GLMM)

'cd /data/project_data/RS_RNASeq/fastq'

my pops

LOL- H 

MMF- C





------
<div id='id-section34'/>   
### Entry 34: 2020-02-27, Thursday.   



------
<div id='id-section35'/>   
### Entry 35: 2020-02-28, Friday.   



------
<div id='id-section36'/>   
### Entry 36: 2020-03-02, Monday.   



------
<div id='id-section37'/>   
### Entry 37: 2020-03-03, Tuesday.   



------
<div id='id-section38'/>   
### Entry 38: 2020-03-04, Wednesday.   



------
<div id='id-section39'/>   
### Entry 39: 2020-03-05, Thursday.   



------
<div id='id-section40'/>   
### Entry 40: 2020-03-06, Friday.   



------
<div id='id-section41'/>   
### Entry 41: 2020-03-09, Monday.   



------
<div id='id-section42'/>   
### Entry 42: 2020-03-10, Tuesday.   



------
<div id='id-section43'/>   
### Entry 43: 2020-03-11, Wednesday.   



------
<div id='id-section44'/>   
### Entry 44: 2020-03-12, Thursday.   



------
<div id='id-section45'/>   
### Entry 45: 2020-03-13, Friday.   



------
<div id='id-section46'/>   
### Entry 46: 2020-03-16, Monday.   



------
<div id='id-section47'/>   
### Entry 47: 2020-03-17, Tuesday.   



------
<div id='id-section48'/>   
### Entry 48: 2020-03-18, Wednesday.   



------
<div id='id-section49'/>   
### Entry 49: 2020-03-19, Thursday.   



------
<div id='id-section50'/>   
### Entry 50: 2020-03-20, Friday.   



------
<div id='id-section51'/>   
### Entry 51: 2020-03-23, Monday.   



------
<div id='id-section52'/>   
### Entry 52: 2020-03-24, Tuesday.   



------
<div id='id-section53'/>   
### Entry 53: 2020-03-25, Wednesday.   



------
<div id='id-section54'/>   
### Entry 54: 2020-03-26, Thursday.   



------
<div id='id-section55'/>   
### Entry 55: 2020-03-27, Friday.   



------
<div id='id-section56'/>   
### Entry 56: 2020-03-30, Monday.   



------
<div id='id-section57'/>   
### Entry 57: 2020-03-31, Tuesday.   



------
<div id='id-section58'/>   
### Entry 58: 2020-04-01, Wednesday.   



------
<div id='id-section59'/>   
### Entry 59: 2020-04-02, Thursday.   



------
<div id='id-section60'/>   
### Entry 60: 2020-04-03, Friday.   



------
<div id='id-section61'/>   
### Entry 61: 2020-04-06, Monday.   



------
<div id='id-section62'/>   
### Entry 62: 2020-04-07, Tuesday.   



------
<div id='id-section63'/>   
### Entry 63: 2020-04-08, Wednesday.   



------
<div id='id-section64'/>   
### Entry 64: 2020-04-09, Thursday.   



------
<div id='id-section65'/>   
### Entry 65: 2020-04-10, Friday.   



------
<div id='id-section66'/>   
### Entry 66: 2020-04-13, Monday.   



------
<div id='id-section67'/>   
### Entry 67: 2020-04-14, Tuesday.   



------
<div id='id-section68'/>   
### Entry 68: 2020-04-15, Wednesday.   



------
<div id='id-section69'/>   
### Entry 69: 2020-04-16, Thursday.   



------
<div id='id-section70'/>   
### Entry 70: 2020-04-17, Friday.   



------
<div id='id-section71'/>   
### Entry 71: 2020-04-20, Monday.   



------
<div id='id-section72'/>   
### Entry 72: 2020-04-21, Tuesday.   



------
<div id='id-section73'/>   
### Entry 73: 2020-04-22, Wednesday.   



------
<div id='id-section74'/>   
### Entry 74: 2020-04-23, Thursday.   



------
<div id='id-section75'/>   
### Entry 75: 2020-04-24, Friday.   



------
<div id='id-section76'/>   
### Entry 76: 2020-04-27, Monday.   



------
<div id='id-section77'/>   
### Entry 77: 2020-04-28, Tuesday.   



------
<div id='id-section78'/>   
### Entry 78: 2020-04-29, Wednesday.   



------
<div id='id-section79'/>   
### Entry 79: 2020-04-30, Thursday.   



------
<div id='id-section80'/>   
### Entry 80: 2020-05-01, Friday.   



------
<div id='id-section81'/>   
### Entry 81: 2020-05-04, Monday.   



------
<div id='id-section82'/>   
### Entry 82: 2020-05-05, Tuesday.   



------
<div id='id-section83'/>   
### Entry 83: 2020-05-06, Wednesday.   



------
<div id='id-section84'/>   
### Entry 84: 2020-05-07, Thursday.   



------
<div id='id-section85'/>   
### Entry 85: 2020-05-08, Friday.   



