---
title: "methylKit_Analysis"
author: "Erika Bueno"
date: "4/5/2020"
output: html_document
---

---
title: "R Notebook"
output: html_notebook
editor_options: 
  chunk_output_type: inline
---


```{r echo=FALSE, message=TRUE, warning=FALSE}
library(methylKit)
library(tidyverse)
library(ggplot2)
library(pheatmap)

# first, we want to read in the raw methylation calls with methylkit

# set directory with absolute path (why is this necessary? I have no idea, but gz files wont work with relative paths)

# getwd()
dir<- "C:/Users/ebuen/OneDrive/GitHub/EcolGenomics"

# read in the sample ids

samples<- read.table("C:/Users/ebuen/OneDrive/GitHub/EcolGenomics/sample_id.txt", header = FALSE)

files<- file.path(dir, samples$V1) #paths to files for both dir and txt
# files path to all coverage files
# all(file.exists(files)




# convert to list
file.list<- as.list(files)
# file.list
# now point to coverage files
# get the names only for naming our samples

nmlist<- as.list(gsub("_1_bismark_bt2_pe.bismark.cov.gz", "", samples$V1))
# use methRead to read in the coverage files

myobj <- methRead(
  location = file.list,
  sample.id = nmlist,
  assembly = "atonsa",
  dbtype = "tabix",
  context = "CpG",
  resolution = "base", #bp res
  mincov = 20, #coverage specific to this study- usually ppl use 10
  treatment = c(0,0,0,0, #treatments- matches to samples to name
                1,1,1,1,
                2,2,2,2,
                3,3,3,3,
                4,4,4,4),
  pipeline = "bismarkCoverage",
  dbdir = "C:/Users/ebuen/OneDrive/GitHub/EcolGenomics/EpiData")

######
# visualize coverage and filter
######

# We can look at the coverage for individual samples with getCoverageStats()
getCoverageStats(myobj[[1]], plot = TRUE) #first entry in myobj

# and can plot all of our samples at once to compare.

# filter samples by depth with filterByCoverage() 
filtered.myobj <- filterByCoverage(myobj,
                                   lo.count = 20, lo.perc = NULL,
                                   high.count= NULL, high.perc = 97.5)

######
# merge samples #only pull ouo sites that are present 
######

#Note! This takes a while and we're skipping it- did not not run this

# use unite() to merge all the samples. We will require sites to be present in each sample or else will drop it

# meth <- unite(filtered.myobj, mc.cores = 3, suffix="united",
#               db.dir= "C:/Users/ebuen/OneDrive/GitHub/EcolGenomics/EpiData")



```


```{r echo=TRUE, message=TRUE, warning=FALSE}
library(methylKit)

meth <- methylKit:::readMethylBaseDB(
        dbpath ="C:/Users/ebuen/OneDrive/GitHub/EcolGenomics/methylBase_united.txt.bgz",
                            dbtype = "tabix",
                            sample.id = unlist(nmlist),
                            assembly = "atonsa", # this is just a string. no actual database
                            context = "CpG",
                            resolution = "base",
                            treatment = c(
                              0,0,0,0,
                              1,1,1,1,
                              2,2,2,2,
                              3,3,3,3,
                              4,4,4,4),
                            destrand = FALSE)

# percMethylation() calculates the percent methylation for each site and sample

pm<- percMethylation(meth)

library(tidyr)
# update.packages("tidyr")
# install.packages("ggplot2")

library(ggplot2)
#plot methylation histograms

ggplot(gather(as.data.frame(pm)), aes(value)) + 
    geom_histogram(bins = 10, color="black", fill="grey") + 
    facet_wrap(~key)

# calculate and plot mean methylation

dim(pm)
sp.means<- colMeans(pm) #mean percent methylation for each treatment group

sample= names(sp.means)
group= substr(sample, 1, 6)
methylation = sp.means

pf<-data.frame(sample, group, methylation)

write.table(pf, file="pf.txt")



p<-ggplot(pf, aes(x=group, y=methylation, color=group)) + stat_summary(color="black") + geom_jitter(width=0.1, size=3) 

print(p)

require(methylKit)
# sample clustering
clusterSamples(meth, dist="correlation", method = "ward.D", plot = TRUE) #no clear clustering by treatment except for F0 generation

# PCA

PCASamples(meth, screeplot=TRUE)
PCASamples(meth, screeplot=FALSE)

# subset with reorganize()
nmlist
#Control vs Amb temp + high CO2
meth_sub_1 <- reorganize(meth, sample.ids = c("AA_F25_1","AA_F25_2",
                                            "AA_F25_3", "AA_F25_4",
                                            "AH_F25_1","AH_F25_2",
                                            "AH_F25_3","AH_F25_4"), 
                       treatment = c(1,1,1,1,2,2,2,2), 
                       save.db = FALSE)

meth_sub_1
# control vs high temp + high C02
meth_sub_2 <- reorganize(meth, sample.ids = c("AA_F25_1","AA_F25_2",
                                            "AA_F25_3", "AA_F25_4",
                                            "HH_F25_1","HH_F25_2",
                                            "HH_F25_3","HH_F25_4"), 
                       treatment = c(1,1,1,1,4,4,4,4), 
                       save.db = FALSE)

meth_sub_2

                             
# calculate differential methylation

myDiff_1 <- calculateDiffMeth(meth_sub_1, overdispersion = "MN", mc.cores = 1, suffix= "AA_AH",
                            adjust = "qvalue",
                            test = "Chisq")

# get all differentially methylated bases

myDiff_1<- getMethylDiff(myDiff_1, qvalue= 0.05, difference=10)


# we can visualize the changes in methylation frequencies quickly.
hist(getData(myDiff_1)$meth.diff)
# get hyper methylated bases

# get hypo methylated bases

#heatmaps first
pm_1 <- percMethylation(meth_sub_1)
# make a dataframe with snp id's, methylation, etc.
sig.in_1 <- as.numeric(row.names(myDiff_1))
pm.sig_1 <- pm_1[sig.in_1,]

 
# add snp, chr, start, stop

din_1 <- getData(myDiff_1)[,1:3]
df.out_1 <- cbind(paste(getData(myDiff_1)$chr, getData(myDiff_1)$start, sep=":"), din_1, pm.sig_1)
colnames(df.out_1) <- c("snp", colnames(din_1), colnames(df.out_1[5:ncol(df.out_1)]))
df.out_1 <- (cbind(df.out_1,getData(myDiff_1)[,5:7]))




require(pheatmap)
my_heatmap_1<- pheatmap(pm.sig_1, show_rownames = FALSE)


ctrmean_1 <- rowMeans(pm.sig_1[,1:4])

h.norm_1<- (pm.sig_1-ctrmean_1)

my_heatmap_1 <- pheatmap(h.norm_1,
        show_rownames = FALSE)

##### if you want to change colors. only because I don't love the default colors.
library(wesanderson)
library(color)
paletteLength <- 50
# myColor <- colorRampPalette(c("cyan1", "black", "yellow1"))(paletteLength)
# myColor<-wes_palette("Zissou1", 49, type = "continuous")

myColor<-colorRampPalette(rev(brewer.pal(n = 7, name ="RdYlBu")))
myBreaks <- c(seq(min(h.norm_1), 0, length.out=ceiling(paletteLength/2) + 1), 
              seq(max(h.norm_1)/paletteLength, max(h.norm_1), length.out=floor(paletteLength/2)))
              
my_heatmap_1<- pheatmap(h.norm_1,
        color=myColor, 
        breaks=myBreaks,
        show_rownames = FALSE)

#####
#let's look at methylation of specific gene or snp
####

df.out
df.plot <- df.out[,c(1,5:12)] %>% pivot_longer(-snp, values_to = "methylation")
df.plot$group <- substr(df.plot$name,1,2)
head(df.plot)

# looking at snp LS049205.1:248
# if you choose a different snp, you can create different plots.

df.plot %>% filter(snp=="LS049205.1:248") %>% 
            ggplot(., aes(x=group, y=methylation, color=group, fill=group)) +
              stat_summary(fun.data = "mean_se", size = 2) +
              geom_jitter(width = 0.1, size=3, pch=21, color="black")

## write bed file for intersection with genome annotation

write.table(file = "C:/Users/ebuen/OneDrive/GitHub/EcolGenomics/diffmeth.bed",
          data.frame(chr= df.out$chr, start = df.out$start, end = df.out$end),
          row.names=FALSE, col.names=FALSE, quote=FALSE, sep="\t")



```
