---
title: "R Notebook"
output: html_notebook
editor_options: 
  chunk_output_type: inline
---

```{r}


# install_github("al2na/methylKit", build_vignettes=FALSE, 
#   repos=BiocManager::repositories(),ref="development",
#   dependencies=TRUE, type="TRUE")
# install.packages("pheatmap")

library(methylKit)
library(pheatmap)


```


```{r}
library(methylKit)
library(tidyverse)
library(ggplot2)
library(pheatmap)

# first, we want to read in the raw methylation calls with methylkit

# set directory with absolute path (why is this necessary? I have no idea, but gz files wont work with relative paths)

# getwd()
dir<- "C:/Users/ebuen/OneDrive/GitHub/EcolGenomics"

# read in the sample ids

samples<- read.table("C:/Users/ebuen/OneDrive/GitHub/EcolGenomics/sample_id.txt", header = FALSE)

files<- file.path(dir, samples$V1) #paths to files for both dir and txt
# files path to all coverage files
# all(file.exists(files)




# convert to list
file.list<- as.list(files)
# file.list
# now point to coverage files
# get the names only for naming our samples

nmlist<- as.list(gsub("_1_bismark_bt2_pe.bismark.cov.gz", "", samples$V1))
# use methRead to read in the coverage files
myobj <- methRead(
  location = file.list,
  sample.id = nmlist,
  assembly = "atonsa",
  dbtype = "tabix",
  context = "CpG",
  resolution = "base", #bp res
  mincov = 20, #coverage specific to this study- usually ppl use 10
  treatment = c(0,0,0,0, #treatments- matches to samples to name
                1,1,1,1,
                2,2,2,2,
                3,3,3,3,
                4,4,4,4),
  pipeline = "bismarkCoverage",
  dbdir = "C:/Users/ebuen/OneDrive/GitHub/EcolGenomics/EpiData")

######
# visualize coverage and filter
######

# We can look at the coverage for individual samples with getCoverageStats()
getCoverageStats(myobj[[1]], plot = TRUE) #first entry in myobj

# and can plot all of our samples at once to compare.

# filter samples by depth with filterByCoverage() 
filtered.myobj <- filterByCoverage(myobj,
                                   lo.count = 20, lo.perc = NULL,
                                   high.count= NULL, high.perc = 97.5)

######
# merge samples #only pull ouo sites that are present 
######

#Note! This takes a while and we're skipping it- did not not run this

# use unite() to merge all the samples. We will require sites to be present in each sample or else will drop it

# meth <- unite(filtered.myobj, mc.cores = 3, suffix="united",
#               db.dir= "C:/Users/ebuen/OneDrive/GitHub/EcolGenomics/EpiData")



```
```{r}

meth <- methylKit:::readMethylBaseDB(
                      dbpath = "/Users/ebuen/OneDrive/GitHub/EcolGenomics/methylBase_united.txt.bgz",
                            dbtype = "tabix",
                            sample.id =   unlist(nmlist),
                            assembly = "atonsa", # this is just a string. no actual database
                            context = "CpG",
                            resolution = "base",
                            treatment =   c(0,0,0,0,
                              1,1,1,1,
                              2,2,2,2,
                              3,3,3,3,
                              4,4,4,4),
                            destrand = FALSE)

#meth

# percMethylation() calculates the percent methylation for each site and sample

pm<- percMethylation(meth)

#plot methylation histograms
p<-ggplot(gather(as.data.frame(pm)), aes(value)) + geom_histogram(bins = 10, color="black", fill="grey") + facet_wrap(~key)

p
# calculate and plot mean methylatio

dim(pm)
sp.means<- colMeans(pm)

p.df <- data.frame(sample=names(sp.means),
          group = substr(names(sp.means), 1,6),
          methylation = sp.means)

ggplot(p.df, aes(x=group, y=methylation, color=group)) + stat_summary(color="black") + geom_jitter(width=0.1, size=3) 


# sample clustering
clusterSamples()

clusterSamples(meth, dist="correlation", method = "ward.D", plot = TRUE)

# PCA
PCASamples()

# subset with reorganize()

meth_sub <- reorganize(meth, sample.ids = c("AA_F00_1","AA_F00_2","AA_F00_3", "AA_F00_4",
                              "HH_F25_1","HH_F25_2","HH_F25_3","HH_F25_4"), treatment = c(0,0,0,0,1,1,1,1),
                       save.db = FALSE)

meth_sub
                             
# calculate differential methylation

myDiff <- calculateDiffMeth(meth_sub, overdispersion = "MN", mc.cores = 1, suffix= "AA_HH",
                            adjust = "qvalue",
                            test = "Chisq")


myDiff<- getMethylDiff(myDiff, qvalue= 0.05, difference=10)

# get all differentially methylated bases



# we can visualize the changes in methylation frequencies quickly.
hist(getData(myDiff)$meth.diff)
# get hyper methylated bases

# get hypo methylated bases

#heatmaps first
pm <- percMethylation(meth_sub)
# make a dataframe with snp id's, methylation, etc.
sig.in <- as.numeric(row.names(myDiff))
pm.sig <- pm[sig.in,]

 

# add snp, chr, start, stop

 pm <- percMethylation(meth_sub)
# make a dataframe with snp id's, methylation, etc.
sig.in <- as.numeric(row.names(myDiff))
pm.sig <- pm[sig.in,]

 

# add snp, chr, start, stop

 

din <- getData(myDiff)[,1:3]
df.out <- cbind(paste(getData(myDiff)$chr, getData(myDiff)$start, sep=":"), din, pm.sig)
colnames(df.out) <- c("snp", colnames(din), colnames(df.out[5:ncol(df.out)]))
df.out <- (cbind(df.out,getData(myDiff)[,5:7]))


my_heatmap<- pheatmap(pm.sig, show_rownames = FALSE)


# get percent methylation matrix

# make a dataframe with snp id's, methylation, etc.

# add snp, chr, start, stop


####
# heatmap
####

my_heatmap <- pheatmap()

# we can also normalize 



#####
#let's look at methylation of specific snps
####

# convert data frame to long form

## write bed file for intersection with genome annotation




```
